processor/runtime/src/main/java/com/redhat/service/smartevents/processor/actions/kafkatopic/KafkaTopicActionInvoker.java

```diff
@@ -27,17 +37,34 @@ public KafkaTopicActionInvoker(Emitter<String> emitter, ProcessorDTO processor,
     }
 
     @Override
-    public void onEvent(String event) {
+    public void onEvent(CloudEvent originalEvent, String transformedEvent) {
+
+        // Extract CE Headers into Kafka Headers.
+        // See https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/bindings/kafka-protocol-binding.md#3231-property-names
+        List<Header> ceHeaders = originalEvent.getAttributeNames()
```

Comment:
Include the CE headers in our message to MC..

This is a bone of contention... perhaps we'd be better off passing our own headers (`rhose-event-id` etc) as the message we send is not a structured CE nor a binary CE.. but we need this meta-data to be able to correlate any DLQ messages with the original CE). However we run the risk of our own headers being removed if they are not _standard_ (IDK.. I might get a chance to test and confirm/deny).